tuxemon/core/states/world/worldstate.py

```diff
@@ -502,13 +502,19 @@ def get_collision_map(self):
         # TODO: overlapping tiles/objects by returning a list
         collision_dict = dict()
 
+        # tile layout must be set first
```

Comment:
why is this changed?

Reply:
I discovered a silent bug in this part of the code as it was interfering with this implementation, was going to comment on it. The tile data contains an NPC stored here as `entity`, in the same data field with parameters such as `enter_from` `exit_to` `continue` as well as the new `blocks` in this PR. The issue was that when the NPC entity is set here, it gets erased by the next call. So if the old `collision_dict[pos]` contained:

`{entity: <npc_object>}`

It was immediately overridden to something like:

`{exit_to: up, enter_from: down}`

My change was to ensure they are mixed instead so no properties are lost:

`{entity: <npc_object>, exit_to: up, enter_from: down}`

From what I gathered this is why it was happening: The `collision_dict.update` call is the one refreshing this field which overrides the old data. It only seems to do that in some circumstances though. But when it does, the `collision_dict[pos] = {"entity": npc}` set just a few lines above is completely replaced, unless that's done afterward and mixed with the old dict.

Reply:
ok fair enough.